SimpleNotePad.java

```diff
@@ -72,142 +65,17 @@ public static void main(String[] args) {
-                return;
-            printerJob.print();
-        } catch (PrinterException pe) {
-            JOptionPane.showMessageDialog(null,
-                    "Printer error" + pe, "Printing error",
-                    JOptionPane.ERROR_MESSAGE);
-        }
-    }
-
-    // Open a selected file using dialog box
-    private void openFile(ActionEvent e) {
-        JFileChooser fc = new JFileChooser();
-        int returnVal = fc.showOpenDialog(null);
-        if (returnVal == JFileChooser.APPROVE_OPTION) {
-            File fileToOpen = fc.getSelectedFile();
-            storeRecent(fileToOpen.getName(), fileToOpen.getAbsolutePath());
-            readFile(fileToOpen);
-        }
-    }
-
-    // Opens recently opened files
-    private void openRecentFile(ActionEvent e) {
-        String filename = e.getActionCommand();   // Get name of file that is selected
-        String pathname = recentFilePath.get(filename);
-        storeRecent(filename, pathname);
-        readFile(new File(pathname));
-    }
-
-    // Store recently opened files in a LinkedHashSet and update menu with 5 most recently opened files
-    private void storeRecent(String filename, String path) {
-        // Add file and pathname to HashMap
-        recentFilePath.put(filename, path);
-        // Add file to ordered LinkedHashSet
-        if (recentFiles.contains(filename)) {
-            recentFiles.remove(filename);   //Repeated file should be put at the top of the list
-        }
-        recentFiles.add(filename);
-
-        // Refresh Recent menu items
-        recent.removeAll();
-
-        // Iterate over last five elements
-        Iterator<String> el = recentFiles.iterator();
-        if (recentFiles.size() > 5) {
-            for (int i = 0; i < recentFiles.size()-5; i++) {
-                el.next();  //skip beginning elements
-            }
-        }
-        // Add last 5 elements to menu
-        while (el.hasNext()) {
-            String name = el.next();
-            JMenuItem item = new JMenuItem(name);
-            recent.add(item, 0);
-            item.addActionListener(this :: openRecentFile);
-        }
-    }
-
-    // Reads file contents and prints to the display
-    private void readFile(File fileToOpen) {
-        display.setText("");
-        Scanner in = null;
-        try {
-            in = new Scanner(fileToOpen);
-            while (in.hasNext()) {
-                String line = in.nextLine();
-                StyledDocument doc = display.getStyledDocument();
-                doc.insertString(doc.getLength(), line + "\n", null);
-            }
-        } catch (Exception ex) {
-            ex.printStackTrace();
-        } finally {
-            in.close();
-        }
-    }
-
-    // Replace selected text with user input (through dialog box)
-    private void replaceText(ActionEvent e) {
-        display.replaceSelection(JOptionPane.showInputDialog(display, "Replace or insert with: "));
-    }
-
-    // Copy selected text
```

Comment:
Nice work making separate methods for the copy and paste functionalities making the code more reusable.